src/game/Strategic/Strategic_Merc_Handler.cc

```diff
@@ -177,6 +177,7 @@ void MercDailyUpdate()
 			if (--s->bCorpseQuoteTolerance < 0) s->bCorpseQuoteTolerance = 0;
 
 			MERCPROFILESTRUCT& p = GetProfile(s->ubProfile);
+			MercProfile const profile(s->ubProfile);
```

Comment:
Does this internalize the check or why isn't it needed here?

Reply:
[GetProfile](https://github.com/ja2-stracciatella/ja2-stracciatella/blob/bee3fe37530d618b7c391bc7e85cf955369a8626/src/game/Tactical/Soldier_Profile.h#L14-L20) checks for a valid profile ID, and aborts on invalid ID.

Reply:
Still quite strict, but it's unlikely to be hit in this place.